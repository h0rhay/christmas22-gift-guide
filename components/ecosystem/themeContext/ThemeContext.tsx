import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
