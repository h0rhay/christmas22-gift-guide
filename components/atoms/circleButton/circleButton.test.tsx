import { render } from '@testing-library/react';
import CircleButton from '.';
import { ThemeProvider } from 'styled-components';
import Theme from 'styles/theme';
import ThemeContext from "components/ecosystem/themeContext/ThemeContext";

describe('CircleButton', () => {

  it('should match snapshot - Text CircleButton', () => {
    const renderComponent = () =>
      render(
        <ThemeProvider theme={Theme}>
          <ThemeContext colorGroup="hero">
            <CircleButton variant='text' text='foobar' clickHandler={() => null} />
          </ThemeContext>
        </ThemeProvider>
      );
    const { container } = renderComponent();
    expect(container).toMatchSnapshot('Text CircleButton');
  });

  it('should match snapshot - Accessibility CircleButton', () => {
    const renderComponent = () =>
      render(
        <ThemeProvider theme={Theme}>
          <ThemeContext colorGroup="hero">
            <CircleButton variant='icon' type='accessibility' clickHandler={() => null} />
          </ThemeContext>
        </ThemeProvider>
      );
    const { container } = renderComponent();
    expect(container).toMatchSnapshot('Accessibility CircleButton');
  });

  it('should match snapshot- Share CircleButton', () => {
    const renderComponent = () =>
      render(
        <ThemeProvider theme={Theme}>
          <ThemeContext colorGroup="hero">
            <CircleButton variant='icon' type='share' clickHandler={() => null} />
          </ThemeContext>
        </ThemeProvider>
      );
    const { container } = renderComponent();
    expect(container).toMatchSnapshot('Share CircleButton');
  });

});
