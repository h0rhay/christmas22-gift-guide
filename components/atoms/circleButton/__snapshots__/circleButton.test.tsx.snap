// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleButton should match snapshot - Accessibility CircleButton: Accessibility CircleButton 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <button
      class="circleButton__Wrapper-sc-o294s7-0 gxqiFO"
    >
      <svg
        fill="currentColor"
        height="25"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`CircleButton should match snapshot - Text CircleButton: Text CircleButton 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <button
      class="circleButton__Wrapper-sc-o294s7-0 gxqiFO"
    >
      <span>
        foobar
      </span>
    </button>
  </div>
</div>
`;

exports[`CircleButton should match snapshot- Share CircleButton: Share CircleButton 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <button
      class="circleButton__Wrapper-sc-o294s7-0 gxqiFO"
    >
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
        />
        <polyline
          points="16 6 12 2 8 6"
        />
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="15"
        />
      </svg>
    </button>
  </div>
</div>
`;
